CONFIG_FORMAT.md

```diff
@@ -0,0 +1,177 @@
+
+**Format**: List of names of required attributes.
+
+**Example**: "lon" : { "required_attributes" : ["axis", "units"] }
+
+## required_intervals
+
+Use this to specify the expected interval between entries in the data. This can
+be used in one of two ways.
+
+1) In the variable itself to define the interval between its entries
+
+This is the simplest usage, for example specifying the number of degrees apart
+entries for lattitude may be.
+
+**Location**: Within a variable's definition
+
+**Format**: Numeric value of expected difference
+
+**Example**: "lat" : { "required_intervals" : 10.0 }
+
+2) Using within a variable to specify the expected interval of an associated
+variable
+
+**Location**: Within a variable's definition
+
+**Format**: Dictionary specifying an associated variable and its required interval
+
+**Example**: "tas" : { "required_intervals" : { "leadtime" : 6.0 } }
+
+This is useful when you want to have more than one variable within a file that
+have the same named associated variables but which may be different for each.
+
+For example you may have be outputting various variables at different timesteps
+depending on what that variable is e.g. tas with a leadtime of 6.0 while tauv
+has a leadtime of 24.
+
+## required_min_max
+
+Use this to specify the expected minimum and maximum values. This is an
+*inclusive* setting so these values must be present. If you want values to fall
+within a specific range then use the "required_range" entry instead.
+
+**Location**: Within a variable's definition
+
+**Format**: Two item list as [min, max]
+
+**Example**: "lon" : { "required_min_max": [0.5, 359.5] }
+
+## required_range
+
+Use this to specify the (inclusive) range within which values must be. If you
+want the minimum and maximum values to be specific values, then use the
+"required_min_max" entry instead.
+
+**Location**: Within a variable's definition
+
+**Format**: Two item list as [min, max]
+
+**Example**: "lon" : { "required_range": [0.0, 360.0] }
+
+## required_values
+
+Use this to specify the values which the varible must have. This is an exact
+matching so, when inspecting the variable, even if all values are present the
+check will still fail if they are not in the expected order.
+
+**Location**: Within a variable's definition
+
+**Format**: List of items in the order you expect them as [item1, item2, ... itemN]
+
+**Example**: "lon" : { "required_values": [0.0, 180.0, 360.0] }
+
+# Other reserved config items
+
+## pattern
+
+Use this to specify the pattern you expect an entry to follow.
+
+**Location**: Within an attribute's definition
+
+**Format**: Python regular expression to match
+
+**Example**: "creation_date": { "pattern": "\\d\\d\\d\\d-\\d\\d-\\d\\d }
+
+## bounds
+
+Use this to specify any associated bounds variables you want to have present in
+your file.
+
+**Location**: Within a variable's definition
+
+**Format**: List of associated bounds names
+
+**Example**: "lon" : ["lon_bnds"]
```

Comment:
Should this be "lon": {"bounds": ["lon_bnds"]} ?

Reply:
Changed